certificate/certificate.go

```diff
@@ -84,15 +87,18 @@ type SubjectPublicKeyInfo struct {
 
 //Currently exporting extensions that are already decoded into the x509 Certificate structure
 type Extensions struct {
-	AuthorityKeyId         string   `json:"authorityKeyId,omitempty"`
-	SubjectKeyId           string   `json:"subjectKeyId,omitempty"`
-	KeyUsage               []string `json:"keyUsage,omitempty"`
-	ExtendedKeyUsage       []string `json:"extendedKeyUsage,omitempty"`
-	SubjectAlternativeName []string `json:"subjectAlternativeName,omitempty"`
-	CRLDistributionPoints  []string `json:"crlDistributionPoint,omitempty"`
-	PolicyIdentifiers      []string `json:"policyIdentifiers,omitempty"`
-	IsNameConstrained      bool     `json:"isNameConstrained,omitempty"`
-	PermittedNames         []string `json:"permittedNames,omitempty"`
+	AuthorityKeyId           string      `json:"authorityKeyId,omitempty"`
+	SubjectKeyId             string      `json:"subjectKeyId,omitempty"`
+	KeyUsage                 []string    `json:"keyUsage,omitempty"`
+	ExtendedKeyUsage         []string    `json:"extendedKeyUsage,omitempty"`
+	SubjectAlternativeName   []string    `json:"subjectAlternativeName,omitempty"`
+	CRLDistributionPoints    []string    `json:"crlDistributionPoint,omitempty"`
+	PolicyIdentifiers        []string    `json:"policyIdentifiers,omitempty"`
+	PermittedDNSDomains      []string    `json:"permittedDNSNames,omitempty"`
+	PermittedIPAddresses     []net.IPNet `json:"permittedIPAddresses,omitempty"`
+	ExcludedDNSDomains       []string    `json:"excludedDNSNames,omitempty"`
+	ExcludedIPAddresses      []net.IPNet `json:"excludedIPAddresses,omitempty"`
```

Comment:
The downside of using `net.IPNet` here is we get a JSON document like this:
```json
    "excludedIPAddresses": [
      {
        "IP": "0.0.0.0",
        "Mask": "AAAAAA=="
      },
      {
        "IP": "::",
        "Mask": "AAAAAAAAAAAAAAAAAAAAAA=="
      }
    ]
```
I don't find this to be very readable or convenient to use. I think we should use an `[]string` instead, and represent them as CIDR, like this:
```json
    "excludedIPAddresses": [
        "0.0.0.0/0",
        "127.0.0.1/16"
    ]
```

We should still store them as IP addresses in database, but expose them as CIDR strings in JSON.